docs/e2e/e2e-checks.md

```diff
@@ -0,0 +1,76 @@
+# End-to-End (E2E) Tests
+
+## Overview
+
+This repository uses [Playwright](https://playwright.dev/) to perform end-to-end (E2E) tests. The tests can be run locally, but also run on [Pull Request preview environments](../infra/pull-request-environments.md). This ensures that any new code changes are validated through E2E tests before being merged.
+
+## Folder Structure
+In order to support e2e for multiple apps, the folder structure will include a base playwright config (`./e2e/playwright.config.js`), and app-specific derived playwright config that override the base config. See the example folder structure below:
+```
+- e2e
+  - playwright.config.js
+  - app/
+    - playwright.config.js
+    - tests/
+      - index.spec.js
+  - app2/
+    - playwright.config.js
+    - tests/
+      - index.spec.js
+```
+
+Some highlights:
+- By default, the base config is defined to run on a minimal browser-set (desktop and mobile chrome)
+- Snapshots will be output locally or in the artifacts of the CI job
+- HTML reports are output to the `playwright-report` folder
+- Parallelism limited on CI to ensure stable execution
+- Accessibility testing can be performed using the `@axe-core/playwright` package (https://playwright.dev/docs/accessibility-testing)
+
+
+## Running Locally
+
+### Running Locally From the Root Directory
+
+Make targets are setup to easily pass in a particular app name and URL to run tests against
+
+```
+make e2e-setup # install playwright deps
+make e2e-test APP_NAME=app BASE_URL=http://localhost:3000 # run tests on a particular app
+```
+
+### Running Locally From the `./e2e` Directory
+
+If you prefer to run package.json run scripts, you can do so by creating a `./e2e/.env` file with an `APP_NAME` and `BASE_URL`
+
+```
+cd e2e
+
+# Create .env file with BASE_URL and APP_NAME
+echo "BASE_URL=http://127.0.0.1:3000" > .env
+echo "APP_NAME=your-app-name" >> .env
+
+npm install
+npm run e2e-test
+```
```

Comment:
do we still need this even though you already defined `baseUrl: baseConfig.use.baseUrl || "localhost:3000"` in the app/playwright.config.js?

Reply:
The intent was to be able to allow them to also run the tests from the `./e2e` folder by passing it an app name .

1) do we want to be able to let them run in the e2e folder? I would vote that it's nice to be able to run it from multiple places but others may think it's more confusing to allow multiple ways to run

2) if we do allow them to run from e2e folder, how do we do that? 
-`.env` is one way
- `cross-env` I believe is another way 
```
 "e2e-test": "cross-env APP_NAME=$npm_config_app_name sh -c 'source .env && npx playwright test --config $APP_NAME/playwright.config.js'"
 
 npm run e2e-test --app_name=app
 ```

Reply:
we should allow it but we should be consistent.

For example:

## Run from e2e

```
# e2e/package.json
"e2e-test": "npx playwright test --config $APP_NAME/playwright.config.js'"

# running
e2e$ APP_NAME=app npm run e2e-test
```

## Run from root

```
# Makefile
e2e-test:
  cd e2e && APP_NAME=$(APP_NAME) npm run e2e-test

# Running
$ make e2e-test APP_NAME=app
```

Reply:
Yea that works! Let's go with that.

wonder if we should put a shell script in front of it to mimic how the make version of the command errors

```
#!/bin/sh

if [ -z "$APP_NAME" ]; then
  echo "Error: APP_NAME environment variable is not set."
  exit 1
fi

npx playwright test --config "$APP_NAME/playwright.config.js"
```

Reply:
sure, except remove the `.sh` extension since we're following [google's shell style guide](https://google.github.io/styleguide/shellguide.html#shell-files-and-interpreter-invocation)

Reply:
Thanks just updated